app/imports/api/smartContract/smartContractDeployment.js

```diff
@@ -0,0 +1,31 @@
+const _ = require('lodash');
+
+export function bothSigned(contract) {
+  const hSignature = contract.homeownerSignature;
+  const tSignature = contract.tenantSignature;
+  const hNotSigned = hSignature === '';
+  const tNotSigned = tSignature === '';
+  console.log(`h: not signed: ${hNotSigned} signature: ${hSignature}`);
+  console.log(`t: not signed: ${hNotSigned} signature: ${hSignature}`);
+  if (hNotSigned || tNotSigned) {
+    return false;
+  }
+  return true;
+}
+
+export function createHomeowner(profiles, hEmail) {
+  const profile = profiles.filter(p => p.owner === hEmail);
+  const homeowner = {};
+  homeowner.address = profile[0].walletAddress;
+  homeowner.privateKey = profile[0].privateKey;
+  return homeowner;
+}
+
+export function createTenant(profiles, tEmail) {
+  const profile = _.filter(profiles, { owner: tEmail });
+  const tenant = {};
+  tenant.address = profile[0].walletAddress;
+  tenant.privateKey = profile[0].privateKey;
+  tenant.period = 'seconds';
+  return tenant;
+}
```

Comment:
From a "coding" perspective, I'd propose the following. (Some of it is purely personal preference I guess, and some of it follows from the DRY principle.)

```
export function createHomeowner(profiles, email) {
  const profile = profile.find(p => p.owner === email);
  return {
    address: profile.walletAddress,
    privateKey: profile.privateKey,
  };
}

export function createTenant(profiles, email) {
  return {
    ...createHomeowner(profiles, email),
   period: 'seconds',
  };
}